File: No_2134/Minimum Swaps to Group All 1's Together II.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        width = nums.count(1)
        ret = length = len(nums)
        
        zeros = nums[:width].count(0)
        for i in range(length):
            ret = min(ret, zeros)
            zeros += nums[(i + width) % length] == 0
            zeros -= nums[i] == 0
            
        return ret

File: No_2134/Minimum Swaps to Group All 1's Together II.cpp
class Solution {
public:
    int minSwaps(vector<int>& nums) {
        int width = count(nums.begin(), nums.end(), 1);
        int ret = nums.size();
        int length = nums.size();
        
        int zeros = count(nums.begin(), nums.begin()+width, 0);
        for (int i = 0; i < length; i++) {
            ret = min(ret, zeros);
            if (nums[(i + width) % length] == 0)
                zeros++;
            if (nums[i] == 0)
                zeros--;
        }
        
        return ret;
    }
};